### ISSUE TITLE
 Got an exception - java.lang.NullPointerException (at line 1) when applying checkstyle containing custom checks

### ISSUE THREAD

**Issue by harcek** [jshiell/checkstyle-idea#164]
Checkstyle containing custom checks returns `java.lang.NullPointerException` (real-time scan enabled, ad hoc inspection does the same).

I add the custom jar in 3rd party checks and enable the checkstyle xml (the jar seems to be loaded well, if it's added within 3rd party checks loading checkstyle complaints about `Unable to instantiate` custom checks). 

![image](https://cloud.githubusercontent.com/assets/656358/7518052/f8f3afa0-f4d8-11e4-981d-6dd6443d5122.png)

It can be seen that once the checkstyle is applied real-time scan ends with a exception

![image](https://cloud.githubusercontent.com/assets/656358/7518611/53a42242-f4dc-11e4-8c78-4ea7b1113a30.png)

Ad hoc inspection throws the same 

![image](https://cloud.githubusercontent.com/assets/656358/7518625/68d29a5e-f4dc-11e4-9bcc-5f8dc5979377.png)

There are no error messages related to this, nothing in jidea log - nothing I could start with. Custom checks and that checkstyle work well in Eclipse and with maven-checkstyle plugin. The jar and checkstyle can be found here https://drive.google.com/folderview?id=0Byssr-kgyjSFflk4MWFjNGVBeUw2SzNjWGJrR204akFxMmt5NFl3Q1oyTkp4TXJLNDBqT0E&usp=sharing. 

Tested on: 
JIDEA 13.1.3 with Checkstyle plugin 4.13.2, JIDEA 14.0.2 with Checkstyle plugin 4.12.0
Custom checkstyle loads checkstyle 6.2. 
```
        <dependency>
            <groupId>com.puppycrawl.tools</groupId>
            <artifactId>checkstyle</artifactId>
            <version>6.2</version>
        </dependency>
```

**Comment by jshiell** [comment]
Thanks for the great report - much appreciated.

I haven't managed to reproduce this on the current version of the plugin (4.15.0 on IDEA 14.1.2, with CS 6.6). I'll keep having a play, but either I'm doing something wrong or it's been fixed either by myself or CheckStyle in the meantime.

4.13.2 used CheckStyle 6.1.1 from memory, due to the JDK7 requirement for 6.2 and above. This wasn't resolved until plugin 4.14.0, which moved to JDK8 and CheckStyle 6.5. This is restricted to IDEA 14.1.1 and above, primarily as OS X users only got a bundled JDK in 14.1 (and prior versions all ran on JDK6). However, I believe this version of the plugin *should* (probably) run on older versions of the IDE, as I don't believe I've made any API usage changes.

So assuming I'm not missing something in my testing (I'll have another go later), I see two solutions:

a) Ideally, update to 14.1.2 and the latest version of the plugin. I appreciate this may not be possible in your situation, as it'll change the CheckStyle version.

b) Edit the plugin JAR and replace the CheckStyle JAR and all dependencies with CheckStyle 6.2. This should work, as long as IDEA is running on JDK7 or 8, as there haven't been any API changes here either. Just be careful with the dependencies.

**Comment by harcek** [comment]
Thanks a lot :).

a) works
b) when compiling my extension with 6.1.1 4.13.2 works as well, I'll create a separate package to cover that for backward compatibility. 

Best
